readthedocs/projects/forms.py

```diff
@@ -493,6 +494,35 @@ def clean_alias(self):
         return alias
 
 
+class ProjectAddonsForm(forms.ModelForm):
+
+    """Form to opt-in into new beta addons."""
+
+    enabled = forms.BooleanField(
+        label="Enable Read the Docs beta addons",
+        help_text="Opt-in into new beta addons",
+        required=False,
+    )
+
+    class Meta:
+        model = Project
+        fields = ()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        enabled = bool(self.instance.addons)
+        self.fields["enabled"].initial = enabled
+
+    def save(self, commit=True):
+        instance = super().save(commit)
+        if self.cleaned_data.get("enabled", False):
+            self.instance.addons = ProjectAddonsConfig.objects.create()
+            self.instance.save()
+        elif self.instance.addons:
+            self.instance.addons.delete()
```

Comment:
Why don't we have an `enabled` field on the actual model, and modify that? This way, if a user ever disables their addons, they lose all their old state, so when they re-enable them, they have to completely recreate eg. their custom docdiff selector? I think keeping that state and just allowing them to enable/disable it seems better.

Reply:
Good point!